transcom/mymove
# E 04823 tickets b 18472 and b 18457 main

## [Integration PR](https://github.com/transcom/mymove/pull/12051)

## [Feature E-04823](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic%3A866910)
- [Frontend: B-18472](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A887939)
- [Backend: B-18457](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A886626)

## Summary

These two tickets introduce new business logic and refactoring with how service member duty location, grade, and weight allotments function within MilMove. At a high level, grade and duty location has moved from the service member to the order. Additionally, grades now reflect a "true" value that is calculated by weight allotment and assigned to a `DBWeightAuthorized`. `DBWeightAuthorized` is directly tied to the maximum billable weight of a move. When a service member grade is updated, it immediately updated the maximum billable weight, whether it be increasing or decreasing the maximum amount. 

## Verification Steps for Reviewers


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test
Frontend:
1. Create a new customer profile and ensure you see the correct weight allowances based on your grade and dependents value. You can see this when adding shipments, as a TOO or services counselor, and throughout the flow.
2. When adding your grade and dependents to the orders, go back and edit them and make sure the weight allowance changes to reflect that.
3. Ensure that Current duty location and Grade are no longer seen in the Service Info or Profile section, and are instead associated with the orders on the review page after adding orders and before submitting the move.

Backend:
1. Login as a TIO
4. Select a move
5. Navigate to move details -> allowances
6. Make note of the "Weight allowance"
7. Update the grade to something that will change the weight allowance (See `pkg/models/entitlements.go`)
8. Review
9. Login as a Counselor
10. Select a move
11. Navigate to move details -> allowances
12. Make note of the "Weight allowance"
13. Update the grade to something that will change the weight allowance (See `pkg/models/entitlements.go`)
14. Review
15. Additionally, you can set the `HasDependents` value on the order to true or false, then update the grade again to review how dependents affect the weight allowance.

## Screenshots
